### PR TITLE
[Prototype] Add Counselor Name to Contact Info tab for Offline Contact

### PR DESCRIPTION
**Pull request by GPaoloni**
Jira: https://bugs.benetech.org/browse/CHI-500
This PR is related to https://github.com/tech-matters/serverless/pull/32

Notes:
- You can point to dev env for serverless in order to test this.
- This PR is better reviewed commit by commit.

This PR:
- Groups functions used to submit contact forms and finish tasks.
- Adds counselor dropdown to offline contact tab, in order to select in behalf of who is being submitted the contact.
- When an offline contact is submitted, a new serverless function (`assignOfflineContact`) is called. The purpose of this function is to create-assign-accept-complete a new Twilio task to the target counselor in order to create the insights for this record. 

The current behavior have some caveats/open questions:
  - The first task created (for the counselor submiting the contact form) also creates some insights. They are "cleaned up" as much as we can but they are still there.
  - The `twilioWorkerId` column of contact stored in the backend is the one of the issuer, not the target counselor. This causes the contact to appear in behalf of the issuer and not the target counselor. For this to be fixed we can either use the target counselor's sid in `twilioWorkerId` (but then the information that this task was submitted by issuer is lost, unless saved inside the raw json or a new column) or change the search query when contacts are offline contacts (this may slow down the query and increase it's complexity). Same happens for cases.

State of insights before and after an offline contact submission with Twitter channel (left is target counselor, right is issuer):
Screenshot from 2021-02-19 15-31-36
Screenshot from 2021-02-19 15-42-58

### CODE REVIEW

**Review by nickhurlburt — APPROVED** [review]
I understand this overall and it makes sense.  Definitely good for @murilovmachado or @andvirga to review in more detail.

File: plugin-hrm-form/src/services/formSumbissionHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+};
+
+export const completeTask = (task: ITask) =>
+  task.attributes.isContactlessTask ? completeContactlessTask(task) : completeContactTask(task);
+
+export const submitContactForm = async (task: ITask, contactForm: Contact, caseForm: Case) => {
+  const { hrmBaseUrl, workerSid, helpline } = getConfig();
+
+  if (task.attributes.isContactlessTask) {
+    const targetSid = 'WK76971332a884bf79ec378f98879d23c2';
```
Comment (andvirga) [inline plugin-hrm-form/src/services/formSumbissionHelpers.ts]: Is this hardcode intentional?

File: plugin-hrm-form/src/services/formSumbissionHelpers.ts [hunk 2]
```diff
@@ -48,17 +48,15 @@ export const completeTask = (task: ITask) =>
   task.attributes.isContactlessTask ? completeContactlessTask(task) : completeContactTask(task);
 
 export const submitContactForm = async (task: ITask, contactForm: Contact, caseForm: Case) => {
-  const { hrmBaseUrl, workerSid, helpline } = getConfig();
+  const { workerSid, helpline } = getConfig();
 
   if (task.attributes.isContactlessTask) {
-    const targetSid = 'WK76971332a884bf79ec378f98879d23c2';
+    const targetSid = contactForm.contactlessTask.createdOnBehalfOf as string;
```
Comment (andvirga) [inline plugin-hrm-form/src/services/formSumbissionHelpers.ts:54]: Ignore my previous comment! 😄

File: plugin-hrm-form/src/services/formSumbissionHelpers.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    } else {
+      await Actions.invokeAction('WrapupTask', { sid, task });
+    }
+  }
+
+  await Actions.invokeAction('CompleteTask', { sid, task });
+};
+
+export const completeContactlessTask = async (task: ITask) => {
+  const { attributes } = task;
```
Comment (murilovmachado) [inline plugin-hrm-form/src/services/formSumbissionHelpers.ts:30]: Much cleaner extracting these functions here! Great!

File: plugin-hrm-form/src/services/formSumbissionHelpers.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+export const submitContactForm = async (task: ITask, contactForm: Contact, caseForm: Case) => {
+  const { hrmBaseUrl, workerSid, helpline } = getConfig();
+
+  if (task.attributes.isContactlessTask) {
+    const targetSid = 'WK76971332a884bf79ec378f98879d23c2';
+    const initialAttributes = { helpline, channelType: 'default', isContactlessTask: true, isInMyBehalf: true };
+    const finalAttributes = buildInsightsData(initialAttributes, contactForm, caseForm, {});
+    const contact = await saveToHrm(task, contactForm, hrmBaseUrl, workerSid, helpline);
+    const _newTask = await assignOfflineContact(targetSid, finalAttributes);
```
Comment (murilovmachado) [inline plugin-hrm-form/src/services/formSumbissionHelpers.ts]: Do we need to store this const `_newTasK`?

File: plugin-hrm-form/src/services/formSumbissionHelpers.ts (hunk 2 above)
Comment (murilovmachado) [reply plugin-hrm-form/src/services/formSumbissionHelpers.ts:54]: I was going to ask the same thing you've asked =D

**Review by murilovmachado — APPROVED** [review]
Looks very good @GPaoloni. This task was very tricky, but you've done really well and pointed out the potential issues that it causes to insights. Great Work!

**Comment by GPaoloni** [comment]
Closed in favor of https://github.com/tech-matters/flex-plugins/pull/374.

**Pull request closed without merging**